extensions/sql-migration/src/wizard/loginMigrationTargetSelectionPage.ts

```diff
@@ -863,23 +877,84 @@ export class LoginMigrationTargetSelectionPage extends MigrationWizardPage {
 				}
 			}));
 
+		// Target port
+		this._targetPortLabel = this._view.modelBuilder.text()
+			.withProps({
+				value: constants.TARGET_PORT_LABEL,
+				description: constants.TARGET_RESOURCE_PORT_INFO,
+				requiredIndicator: true,
+				CSSStyles: { ...styles.LABEL_CSS, 'margin-top': '-1em' }
+			}).component();
+
+		this._targetPortInputBox = this._view.modelBuilder.inputBox()
+			.withProps({
+				width: '300px',
+				inputType: 'text',
+				placeHolder: constants.TARGET_PORT_PLACEHOLDER,
+				value: this._defaultSqlPort,
+				required: true,
+				CSSStyles: { 'margin-top': '-1em' },
+			})
+			.withValidation((inputBox) => {
+				let value = inputBox.value ?? "";
+				this._validatePort(value);
+				return this._isPortValid;
+			})
+			.component();
+
+		this._disposables.push(
+			this._targetPortInputBox.onTextChanged(
+				(value: string) => {
+					this.migrationStateModel._targetPort = value ?? "";
+					this._updateConnectionButtonState();
+				}));
+
+		this._disposables.push(
+			this._targetPortInputBox.onValidityChanged(
+				valid => this._updateConnectionButtonState()));
+
 		return this._view.modelBuilder.flexContainer()
 			.withItems([
 				this._azureResourceGroupLabel,
 				this._azureResourceGroupDropdown,
 				this._azureResourceDropdownLabel,
-				this._azureResourceDropdown])
+				this._azureResourceDropdown,
+				this._targetPortLabel,
+				this._targetPortInputBox,])
 			.withLayout({ flexFlow: 'column' })
 			.component();
 	}
 
+	/**
+	 * The function validates the port number
+	 * @param port Port number
+	 */
+	private _validatePort(port: string): void {
+		// check if port is empty
+		if (port === "") {
+			this._isPortValid = false;
+			return;
+		}
+
+		// Check if the port is a number, port is an integer and port is in the range of 0 to 65535
+		let portNumber = Number(port);
+		if (isNaN(portNumber) || !Number.isInteger(portNumber) || portNumber < 0 || portNumber > 65535) {
```

Comment:
you might be able to narrow you valid ranges down more I use this range for Oracle
export const OraclePortMinimum = 1024;
export const OraclePortMaximum = 65536;

Reply:
Keeping it 0 to 65535 for now, considering we want to provide generic range, and it should be able to take care of future changes as suggested by PM's.